BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -540,13 +540,15 @@ - (void)processResponse:(NSURLResponse *)response
 
 - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)response error:(NSError *)error
 {
-    if (self.HTTPResponse == nil) {
+    if (self.HTTPResponse == nil && response != nil) {
         // If the response has not already been handled (some, but not all, operations get
-        // 'processResponse:' earlier in the process) handle it here.
+        // 'processResponse:' earlier in the process), handle it here.
+        // Also, do not handle a nil response. That could happen if there was a failure on the whole
+        // network connection, for example. In that case, there is no response.
         [self processResponse:response];
     }
 
-    if (data != nil) {
+    if ([data length] > 0) {
```

Comment:
Yeah, I'll add a comment. Basically the data gets initialized to [NSData data] which is non-nil empty data. If at the end it is still empty, we do not want to process the data because that could result in a false-negative error (like being unable to parse JSON from the data). Especially for JSON operations this is common because the network data gets stored in a separate variable and the data is left as-is.